src/app/components/fairChain/fairChain.component.ts

```diff
@@ -320,23 +259,11 @@ export class FairChainComponent implements OnInit {
    * Downloads the file as Graph.json with the method in importExport.service.
    */
   public exportGraph(){
-    this.updateNodePositions();
-    var text = "{\"nodes\":" + JSON.stringify(this.nodes) +",\"edges\":" + JSON.stringify(this.edges)+"}";
+    var text = this.importExportService.convertNetworkToJSON(this.nodes, this.edges);
     var filename = "Graph.json";
     this.importExportService.download(filename, text);
```

Comment:
Maybe we could just replace a lot of this by making this.importExportService.download(this.nodes, this.edges), then it covertes it to JSON inside and also the file name  can be written inside ExportExportServie